lib/matplotlib/axes/_axes.py

```diff
@@ -36,6 +36,25 @@
 from matplotlib.axes._base import _AxesBase, _process_plot_format
 from matplotlib.axes._secondary_axes import SecondaryAxis
 
+try:
+    from numpy.lib.histograms import histogram_bin_edges
+except ImportError:
+    def histogram_bin_edges(arr, bins, range=None, weights=None):
+        if isinstance(bins, str):
+            # rather than backporting the internals, just do the full
+            # computation.  If this is too slow for users, they can
+            # update numpy, or pick a manual number of bins
+            return np.histogram(arr, bins, range, weights)[1]
+        else:
+            if bins is None:
+                # hard-code numpy's default
+                bins = 10
+            if range is None:
+                range = np.nanmin(arr), np.manmax(arr)
```

Comment:
Note that numpy does not use the nan variants here.

Reply:
Agreed that if we want to support nans, we should pre-remove them ourselves before passing them to histogram_bin_edges.

Reply:
... also not the typo here....

Reply:
Just took me about 15 minutes to sort out what the typo was....